From 2d86fbb556ef413c57cb0bfbfbfbd5ee8b04e52a Mon Sep 17 00:00:00 2001
From: Christoph Berg <myon@debian.org>
Date: Sun, 16 Feb 2020 22:41:59 +0100
Subject: [PATCH] Set SOVERSION on library, and install to default lib dir

CMakeLists.txt was erroneously setting the SOVERSION on the binary
instead of the library. Also, properly install to
/usr/lib/<archtriplet>.
---
 CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f40a057..27fd193 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -158,6 +158,7 @@ endif()
 add_library(dsdcc SHARED
     ${dsdcc_SOURCES}
 )
+set_target_properties(dsdcc PROPERTIES VERSION ${VERSION} SOVERSION ${MAJOR_VERSION})
 
 if (USE_MBELIB AND LIBMBE_FOUND)
     target_link_libraries(dsdcc ${LIBMBE_LIBRARY})
@@ -183,8 +184,6 @@ add_executable(dsdccx
     dsd_main.cpp
 )
 
-set_target_properties(dsdccx PROPERTIES VERSION ${VERSION} SOVERSION ${MAJOR_VERSION})
-
 target_include_directories(dsdccx PUBLIC
     ${PROJECT_SOURCE_DIR}
     ${CMAKE_CURRENT_BINARY_DIR}
@@ -222,5 +221,5 @@ INSTALL(
 if(BUILD_TOOL)
     install(TARGETS dsdccx DESTINATION bin)
 endif(BUILD_TOOL)
-install(TARGETS dsdcc DESTINATION lib${LIB_SUFFIX})
+install(TARGETS dsdcc DESTINATION ${LIB_INSTALL_DIR})
 install(FILES ${dsdcc_HEADERS} DESTINATION include/${PROJECT_NAME})
